docs/configuration.asciidoc

```diff
@@ -15,20 +13,334 @@ server_url: 'http://localhost:8200'
+of traffic.
+
+[float]
+[[config-flush-interval]]
+==== `flush_interval`
+
+|============
+| Environment                  | `Config` key     | Default
+| `ELASTIC_APM_FLUSH_INTERVAL` | `flush_interval` | `10`
+|============
+
+Interval with which transactions should be sent to the APM server, in seconds.
+A lower value will increase the load on your APM server,
+while a higher value can increase the memory pressure on your app.
+A higher value also impacts the time until transactions are indexed and searchable
+in Elasticsearch.
+
+[float]
+[[config-transaction-sample-rate]]
+==== `transaction_sample_rate`
+
+|============
+| Environment                           | `Config` key              | Default
+| `ELASTIC_APM_TRANSACTION_SAMPLE_RATE` | `transaction_sample_rate` | `1.0`
+|============
+
+By default, the agent will sample every transaction (e.g. request to your service).
+To reduce overhead and storage requirements, you can set the sample rate to a value
+between `0.0` and `1.0`.
+We still record overall time and the result for unsampled transactions, but no
+context information, tags, or spans.
+
+[float]
+[[config-transaction-max-spans]]
+==== `transaction_max_spans`
+
+|============
+| Environment                         | `Config` key            | Default
+| `ELASTIC_APM_TRANSACTION_MAX_SPANS` | `transaction_max_spans` | `500`
+|============
+
+Limits the amount of spans that are recorded per transaction.
+This is helpful in cases where a transaction creates a very high amount of spans
+(e.g. thousands of SQL queries).
+Setting an upper limit will prevent overloading the agent and the APM server with
+too much work for such edge cases.
+
+[float]
+[[config-verify-server-cert]]
+==== `verify_server_cert`
+|============
+| Environment                       | `Config` key         | Default
+| `ELASTIC_APM_VERIFY_SERVER_CERT`  | `verify_server_cert` | `true`
+|============
+
+By default, the agent verifies the SSL certificate if you use an HTTPS connection to
+the APM server.
+Verification can be disabled by changing this setting to `false`.
+
+[float]
+[[config-disabled-spies]]
+==== `disabled_spies`
+
+[options="header"]
+|============
+| Environment                  | `Config` key     | Default
+| `ELASTIC_APM_DISABLED_SPIES` | `disabled_spies` | `['json']`
+|============
+
+Elastic APM automatically instruments select third party libraries.
+Use this option to disabled any of these.
+
+[float]
+[[config-custom-header-filters]]
+==== `custom_key_filters`
+[options="header"]
+|============
+| Environment                      | `Config` key         | Default | Example
+| `ELASTIC_APM_CUSTOM_KEY_FILTERS` | `custom_key_filters` | `[]`    | `['MyAuthHeader']`
+|============
+
+Elastic APM strips
+https://github.com/elastic/apm-agent-ruby/blob/1.x/lib/elastic_apm/filters/secrets_filter.rb[
+what looks like confidential information] from the request/response headers.
```

Comment:
> from the request/response headers, params

Reply:
Until we actually do strip them from the params, I want to keep it to the headers.